src/main/java/com/redhat/cloud/notifications/routers/JsonParseExceptionMapper.java

```diff
@@ -0,0 +1,18 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.fasterxml.jackson.core.JsonParseException;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+
+@Provider
+public class JsonParseExceptionMapper implements ExceptionMapper<JsonParseException> {
+
+    @Override
+    public Response toResponse(JsonParseException exception) {
+        return Response.status(BAD_REQUEST).entity(exception.getMessage()).build();
```

Comment:
@theute Returning the exception message will help the caller fix the request, but also show we're deserializing the body with Jackson. This could be considered a security weakness, but since we're talking about an open-source project, I'm not sure this is a problem here.

Reply:
And you need to be Red Hat employee to have access to this API

Reply:
The mapper will be used for all APIs, internal and external. But I still think it's ok.